src/extensions/client/mod.rs

```diff
@@ -176,61 +182,71 @@ impl Client {
                                 return;
                             }
 
-                            let result = ws.request(&method, params.clone()).await;
-                            match result {
-                                result @ Ok(_) => {
-                                    request_backoff_counter.store(0, std::sync::atomic::Ordering::Relaxed);
-                                    // make sure it's still connected
-                                    if response.is_closed() {
-                                        return;
+                            if let Ok(result) =
```

Comment:
this background task code is slowing getting out of control... we should seek ways to refactor this gigantic handler to something more manageable later

Reply:
agree